packages/core/signal/src/core.ts

```diff
@@ -239,3 +232,26 @@ export function _setSignalProvider<SignalName extends keyof AlwatrRequestSignals
     {receivePrevious: options?.receivePrevious ?? true},
   );
 }
+
+/**
+ * Remove a signal provider
+ *
+ * Example:
+ *
+ * ```ts
+ * _removeSignalProvider('content-change');
+ * ```
+ */
+export function _removeSignalProvider<SignalName extends keyof AlwatrRequestSignals>(
+    signalName: SignalName,
+): void {
+  logger.logMethodArgs('_removeSignalProvider', {signalName});
+
+  const signal = __getSignalObject(`request-${signalName}` as unknown as SignalName);
+  if (signal.listenerList.length > 0) {
+    logger.accident('setSignalProvider', 'signal_provider_already_set', 'another provider defined and will removed', {
```

Comment:
Its not an accident inside _removeSignalProvider!

Reply:
OK